main.c

```diff
@@ -287,12 +287,17 @@ int main(int argc, char *argv[])
             } else if (i == 1) {
                 timer_event();
             } else if (i == 2) {
+                if (fds[i].revents & (POLLERR | POLLHUP)) {
+                    goto quit;
+                }
                 wl_display_dispatch(wl_display);
```

Comment:
Can we test return value of `wl_display_dispatch` instead? It should return `-1` on failure.

Reply:
Makes sense